third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -623,6 +623,26 @@ func resourceContainerCluster() *schema.Resource {
 			},
 <% end -%>
 
+<% unless version == 'ga' -%>
+			"confidential_nodes": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				MaxItems:    1,
+				Description: `Configuration for the confidential nodes feature, which makes nodes run on confidential VMs. Warning: This configuration can't be changed (or added/removed) after cluster creation without deleting and recreating the entire cluster.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:        schema.TypeBool,
+							Required:    true,
```

Comment:
I believe this field should be ForceNew as well. Right now the schema says this field is updatable, but there is no logic in the update method that does anything with it. Resulting in no change for your test

Reply:
Yep, that seems to have been the issue - thanks!